android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java

```diff
@@ -131,8 +149,10 @@ public void onCapabilitiesChanged(
 
         @Override
         public void onLinkPropertiesChanged(Network network, LinkProperties linkProperties) {
-            mNetwork = network;
-            mNetworkCapabilities = getConnectivityManager().getNetworkCapabilities(network);
+            if (mNetwork != null) {
```

Comment:
I'm not sure I understand what the purpose of this null check is. Wouldn't this mean that whatever the `mNetwork` value is first set to would always be used. Did you maybe mean to do a null check on `network` instead?

Reply:
This null check is because in Android 10, onLinkPropertiesChanged is being called when the network is already lost or unavailable. This results in a second notification with incorrect details being sent from this library when the network was lost / suspended

Because the onLost and onUnavailable handlers above are setting mNetwork to null this check will keep the second (erroneous) event from being triggered if the network is already lost or unavailable.

Reply:
I've left a comment in the method for clarity.

Reply:
Also verified with an Android 9 device that this is backward compatible :)

Reply:
Sounds good!